File: programs/SelectionSort.java
public class SelectionSort {
    public static void main(String[] args) {
        int[] arr = { 32, 99, 54, 9, 67, 38, 93 };
        int count=0;
        for (int i = 0; i < arr.length - 1; i++) {
            int minIdx = i;
            for (int j = i + 1; j < arr.length; j++) {
                if (arr[j] < arr[minIdx]) {
                    minIdx = j;
                }
            }
            count++;
            // Swap smallest found element with current position
            int temp = arr[minIdx];
            arr[minIdx] = arr[i];
            arr[i] = temp;
        }
         System.out.println(count);
        // Print array
        for (int num : arr)
            System.out.print(num + " ");

       
    }
}

File: python-programs/SelectionSort.py
def selection_sort(arr):
    for i in range(len(arr)-1):
        minIdx = i
        for j in range(len(arr)):
            if(arr[j] < arr[minIdx]):
                minIdx = j 

        temp = arr[minIdx]
        arr[minIdx] = arr[i]
        arr[i] = temp

    return arr

def main():
    arr = [3,6,1,5,8,3,8,1,10,4]
    print(selection_sort(arr))

    main()